### PR TITLE
dockerized the ACS code

### PR DESCRIPTION
**Pull request by jtwalsh0**
The dockerfile builds an image, downloads all the code and data for a given state, processes the data, then copies the data to the database

### CODE REVIEW

**Review by jesteria — COMMENTED** [review]
😎

**Comment by potash** [comment]
Any chance you can leave the pure drake instructions in the README, even if buried in a second section? The lead ETL pipeline links to it and in the past DoIT have shied away from docker.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:14.04
+
+RUN apt-get update && \
+    apt-get install -y wget \
+                       curl \
+                       software-properties-common \
```
Comment (nanounanue) [inline Dockerfile]: What is this for?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                       git \
```
Comment (nanounanue) [inline Dockerfile]: This is not needed (see comment below)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                       unzip \
+                       postgresql \
+                       default-jdk
```
Comment (nanounanue) [inline Dockerfile]: If you need `jdk` you can use one of the dsapp `docker` images located in the `ECR`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+RUN add-apt-repository ppa:ubuntugis/ppa && apt-get update && apt-get install -y gdal-bin
+
+RUN curl -o /bin/drake https://raw.githubusercontent.com/Factual/drake/master/bin/drake
+RUN chmod 755 /bin/drake
+CMD ["drake"]
+
+WORKDIR /
+RUN git clone https://github.com/dssg/acs2pgsql.git
```
Comment (nanounanue) [inline Dockerfile]: Download an specific version, do the same than in line `14`

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:14.04
+
+RUN apt-get update && \
+    apt-get install -y wget \
+                       curl \
```
Comment (nanounanue) [inline Dockerfile]: If you already have `wget`, why do you need `curl`?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+RUN curl -o /bin/drake https://raw.githubusercontent.com/Factual/drake/master/bin/drake
+RUN chmod 755 /bin/drake
+CMD ["drake"]
+
+WORKDIR /
+RUN git clone https://github.com/dssg/acs2pgsql.git
+COPY default_profile /acs2pgsql/default_profile
+
+WORKDIR /acs2pgsql
+RUN SHELL=/bin/bash drake --auto
```
Comment (nanounanue) [inline Dockerfile]: Maybe you want to use `CMD` here

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+                       git \
+                       unzip \
+                       postgresql \
+                       default-jdk
+
+RUN add-apt-repository ppa:ubuntugis/ppa && apt-get update && apt-get install -y gdal-bin
+
+RUN curl -o /bin/drake https://raw.githubusercontent.com/Factual/drake/master/bin/drake
+RUN chmod 755 /bin/drake
+CMD ["drake"]
```
Comment (nanounanue) [inline Dockerfile]: Maybe this is an `ENTRYPOINT`

File: Dockerfile (hunk 5 above)
Comment (jtwalsh0) [reply Dockerfile]: :(

File: Dockerfile (hunk 1 above)
Comment (jtwalsh0) [reply Dockerfile]: gdal requires it

File: Dockerfile (hunk 2 above)
Comment (jtwalsh0) [reply Dockerfile]: the underlying code uses git. maybe we should change that.

File: Dockerfile [hunk 8]
```diff
@@ -1,23 +1,27 @@
 FROM ubuntu:14.04
 
+ENV SHELL="/bin/bash"
```
Comment (jesteria) [inline Dockerfile]: Just curious, which part of this required a bash shell?

File: Dockerfile [hunk 9]
```diff
@@ -1,23 +1,27 @@
...
 RUN apt-get update && \
-    apt-get install -y wget \
-                       curl \
-                       software-properties-common \
+    apt-get install -y default-jdk \
                        git \
-                       unzip \
                        postgresql \
-                       default-jdk
+                       unzip \
```
Comment (jesteria) [inline Dockerfile]: Really doesn't make a difference, but if you like, the Github API also offers tarball archives, (and I think you can safely expect `tar` to be pre-installed), _e.g._ –

    wget -O - https://api.github.com/repos/dssg/acs2pgsql/tarball/d25c8dc7c231645f70a36b4e1fd4466c7e9a8ed1 | tar -xzf -

(You might want `tar --strip-components=1 …` to remove an encapsulating directory though I don't think so.)

Alternatively, so long as you've got Git, `git archive` works well –

    git archive --remote=git@github.com:dssg/acs2pgsql.git d25c8dc7c231645f70a36b4e1fd4466c7e9a8ed1 | tar -xf -

File: Dockerfile (hunk 8 above)
Comment (jtwalsh0) [reply Dockerfile]: The Drakefile uses bash syntax, so drake fails

File: Dockerfile (hunk 9 above)
Comment (jtwalsh0) [reply Dockerfile]: the drake script calls unzip, so I need to install it anyway (without more substantial refactoring)

**Pull request merged**